### PR TITLE
Implement the POC

### PR DESCRIPTION
**Pull request by NoriSte**
All the info is contained in the README.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,21 +1,319 @@
...
 
-Run `nx serve feature-first-hasura-console-poc` for a dev server. Navigate to http://localhost:4200/. The app will automatically reload if you change any of the source files.
 
-## Understand this workspace
 
-Run `nx graph` to see a diagram of the dependencies of the projects.
+<details id="context">
+  <summary><b>Context</b></summary>
+
+1. The Hasura Console can be loaded in six different "types", and two different "modes" (12 cases to manage). This is messy represented by the `window.__env` object the Hasura finds when loads.
```
Comment (nicoinch) [inline README.md]: **typo:** Missing word *the Hasura **Console** finds when loads*.

File: README.md [hunk 2]
```diff
@@ -1,21 +1,319 @@
...
+2. When the Console is launched by the CLI server, we must care about [the old CLI server and the new one](https://github.com/hasura/graphql-engine-mono/pull/7256#discussion_r1071239279) (the cases to manage grows to 18).
+3. There are additional things to consider that are retrieved dynamically
+   1. Lux entitlements
+   2. Pricing tiers
+   3. EE Trial license
+   4. ... and that's just the beginning...
+
+*(here is a [good summary by @beaussan](https://hasurahq.slack.com/archives/C0423EHN686/p1676297462152259?thread_ts=1676291643.033329&cid=C0423EHN686) about the current situation).*
+
+All of the abovementioned points concur in showing some features or not, in trying to upsell Hasura to the customers, etc.
```
Comment (nicoinch) [inline README.md]: **typo:** abovementioned => above mentioned

File: README.md [hunk 3]
```diff
@@ -1,21 +1,319 @@
...
+<details id="whats-the-platform-team-goal">
+  <summary><b>What's the Platform team goal?</b></summary>
+
+Essentially, to ease the developer life dealing with the above mess. This is possible through
+
+1. Centralizing manage all the data/info/properties/vars that impact what the Console shows
+2. Allow this centralized management to scale for future needs
+3. Hide all the ~mess~ **implementation details** of dealing with the `window.__env`, pricing plans, entitlements, etc.
+
+Steps and scope of this POC
```
Comment (nicoinch) [inline README.md]: **question:** Should it be a title?

File: README.md [hunk 4]
```diff
@@ -1,21 +1,319 @@
...
+
+## What to expect from this POC
+
+This POC **does not include**:
+- battle-tested code
+- Storybook utilities
+- definitive libraries and file system
+- some thorough real use cases the Console must deal with
+
+This POC **includes**:
```
Comment (nicoinch) [inline README.md:74]: **suggestion:** Maybe beginning with what is included eases readability.

File: README.md [hunk 5]
```diff
@@ -1,21 +1,319 @@
...
+- a high-level proposal of the TS APIs needed to hide/show a feature
```
Comment (nicoinch) [inline README.md]: **suggestion:** Maybe you should not use the *high-level* word as it suggests the API should be refined.

File: README.md [hunk 6]
```diff
@@ -1,21 +1,319 @@
...
+5. `useIsFeatureEnabled` is reactive, so for instance when an EE trial license is activated, `useIsFeatureEnabled` makes the React components consuming it re-render
+
+
+**I see Rect APIs, but I do not see pure JavaScript APIs, why?**
+
+React APIs includes "reactivity" by definition. Vanilla JavaScript APIs cannot offer the same reactivity in an easy way. If you need to consume the Hasura plan data offered by `useIsFeatureEnabled`, read it from a React component and pass it down to your vanilla JavaScript functions.
+
+**How will I be able to access `window.__env` if the goal of this POC is also to stop accessing it?**
+
+We will maybe expose a `useEnvVars_UNSECURE` hook and we will look at whe/where is needed.
```
Comment (nicoinch) [inline README.md]: **typo:** Missing letter *whe**n**/where*

File: libs/hasura-features/src/lib/compatibility.ts [hunk 7]
```diff
@@ -0,0 +1,227 @@
...
+  EELiteLicense,
+  CliMode,
+} from './store';
+
+type EnabledOrNot = 'enabled' | 'disabled';
+type RequiredOrNot = 'required' | 'notRequired';
+type CliModeCompatibility = 'cliOnly' | 'serverOnly' | 'cliOrServer';
+
+// Everything at the root is interpreted like an OR operator: {ce:'enabled', eeLiteWithLicense:'enabled'} means "enabled if ce OR eeLiteWithLicense".
+// ALl the properties are required on purpose: because it's too easy to forget to add a check, especially when we add new conditions.
```
Comment (nicoinch) [inline libs/hasura-features/src/lib/compatibility.ts]: **typo:** ALl => All

File: libs/hasura-features/src/lib/compatibility.ts [hunk 8]
```diff
@@ -0,0 +1,227 @@
...
+    NeonDatabaseIntegration: RequiredOrNot;
+    DatadogIntegration: RequiredOrNot;
+  };
+
+  // While the other checks act as an OR operator, the eeLiteLicense acts as an AND operator to the 'eeLite' check
+  eeLite: EnabledOrNot;
+  eeLiteLicense: RequiredOrNot;
+};
+
+// A utility object mostly used for testing
```
Comment (nicoinch) [inline libs/hasura-features/src/lib/compatibility.ts]: **suggestion:** Why not moving it to a test file?

File: apps/feature-first-hasura-console-poc/src/app/Console.tsx [hunk 9]
```diff
@@ -0,0 +1,29 @@
+import { useConsoleInfoStore } from '@feature-first-hasura-console-poc/hasura-features';
+import { Neon } from './features/Neon';
+import { Prometheus } from './features/Prometheus';
+
+export function Console() {
+  const state = useConsoleInfoStore((state) => state);
+
+  return (
+    <>
+      <br />
```
Comment (nicoinch) [inline apps/feature-first-hasura-console-poc/src/app/Console.tsx:10]: **joke:** Were you cold @NoriSte? (BrrrBrrrrBrrr)

**Review by nicoinch — APPROVED** [review]
Great implementation! I hope I will be able to play with it soon! (just some typo)

File: README.md (hunk 1 above)
Comment (NoriSte) [reply README.md]: Done 😊

File: README.md (hunk 2 above)
Comment (NoriSte) [reply README.md]: Done 😊

File: README.md (hunk 3 above)
Comment (NoriSte) [reply README.md]: Removed, it was a leftover

File: README.md (hunk 4 above)
Comment (NoriSte) [reply README.md:74]: Done 😊

File: README.md (hunk 5 above)
Comment (NoriSte) [reply README.md]: Done 😊

File: README.md (hunk 6 above)
Comment (NoriSte) [reply README.md]: Done 😊

File: libs/hasura-features/src/lib/compatibility.ts (hunk 7 above)
Comment (NoriSte) [reply libs/hasura-features/src/lib/compatibility.ts]: Done 😊

File: libs/hasura-features/src/lib/compatibility.ts (hunk 8 above)
Comment (NoriSte) [reply libs/hasura-features/src/lib/compatibility.ts]: Done 😊

File: apps/feature-first-hasura-console-poc/src/app/Console.tsx (hunk 9 above)
Comment (NoriSte) [reply apps/feature-first-hasura-console-poc/src/app/Console.tsx:10]: 😂😂😂

**Pull request merged**